### PR TITLE
Refined DF Calculation in DDP for more general Expression for addressing variable length stages

### PR DESCRIPTION
**Pull request by sambuddhac**
## Description

This PR refines the evaluation of discount factor (DF) calculation in DDP making it more generic for variable length stages

## What type of PR is this? (check all applicable)

- [x] Bug Fix
- [x] Documentation Update

## Related Tickets & Documents

Fixes Issue #671 
https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword 

## Checklist

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and .md files under /docs/src have been updated if necessary.
- [x] The latest changes on the target branch have been incorporated, so that any conflicts are taken care of before merging. This can be accomplished either by merging in the target branch (e.g. 'git merge develop') or by rebasing on top of the target branch (e.g. 'git rebase develop'). Please do not hesitate to reach out to the GenX development team if you need help with this.
- [x] Code has been tested to ensure all functionality works as intended.
- [x] CHANGELOG.md has been updated (if this is a 'notable' change).
- [x] I consent to the release of this PR's code under the GNU General Public license.

## How this can be tested

Tested it against 6_three_zones_with_multistage

## Post-approval checklist for GenX core developers
After the PR is approved

- [x] Check that the latest changes on the target branch are incorporated, either via merge or rebase
- [ ] Remember to squash and merge if incorporating into develop

### CODE REVIEW

File: src/multi_stage/dual_dynamic_programming.jl [hunk 1]
```diff
@@ -617,6 +617,11 @@ returns: JuMP model with updated objective function.
 """
 function initialize_cost_to_go(settings_d::Dict, EP::Model, inputs::Dict)
     cur_stage = settings_d["CurStage"] # Current DDP Investment Planning Stage
+    stage_len_1 = 0
```
Comment (lbonaldo) [inline src/multi_stage/dual_dynamic_programming.jl:620]: You could rename `stage_len_1` with something more descriptive, e.g.,: 
```suggestion
    cum_years = 0
```

File: src/multi_stage/dual_dynamic_programming.jl [hunk 2]
```diff
@@ -617,6 +617,11 @@ returns: JuMP model with updated objective function.
...
+    for stage_count in 1:(cur_stage-1)
+        stage_len_1 += settings_d["StageLengths"][stage_count]
+    end
+    println("Value of stage len_1 in stage ", cur_stage, " is ", stage_len_1)
```
Comment (lbonaldo) [inline src/multi_stage/dual_dynamic_programming.jl:624]: Probably this print statement is no longer necessary.
```suggestion
```

File: src/multi_stage/write_multi_stage_costs.jl [hunk 3]
```diff
@@ -46,6 +52,7 @@ function write_multi_stage_costs(outpath::String, settings_d::Dict, inputs_dict:
 
     # Remove "cTotal" from results (as this includes Cost-to-Go)
     df_costs = df_costs[df_costs[!, :Costs] .!= "cTotal", :]
+    println("**WARNING** : The cost calculation of the multi-stage GenX is approximate currently, and we will be refining it more in one of the future releases.")
```
Comment (lbonaldo) [inline src/multi_stage/write_multi_stage_costs.jl:55]: You could also use the `@warn` macro to issue a warning together with or in place of the print statement 
```suggestion
    @warn("The cost calculation of the multi-stage GenX is approximate currently, and we will be refining it more in one of the future releases.")
```

**Review by lbonaldo — APPROVED** [review]
Thank you @sambuddhac! Approved it with some comments.

**Comment by sambuddhac** [comment]
> Thank you @sambuddhac! Approved it with some comments.

Thanks, @lbonaldo . I have incorporated all your suggestions. And have made sure to check the integrity of the code. Now, merging this to release branch.

**Pull request merged**